aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/LocalizedSamplingStrategy.java

```diff
@@ -77,54 +97,52 @@ private SamplingRuleManifest getRuleManifest(URL ruleLocation) {
         }
     }
 
-    private SamplingRuleManifest getDefaultRuleManifest() {
+    private static SamplingRuleManifest getDefaultRuleManifest() {
         try {
             return MAPPER.readValue(LocalizedSamplingStrategy.DEFAULT_RULES, SamplingRuleManifest.class);
         } catch (IOException ioe) {
             throw new RuntimeException("Unable to parse " + LocalizedSamplingStrategy.DEFAULT_RULES + ".", ioe);
         }
     }
 
-    private void processRuleManifest(SamplingRuleManifest manifest) {
-        if (manifest == null) {
-            return;
-        }
-        defaultRule = manifest.getDefaultRule();
-        if (!supportedVersions.contains(manifest.getVersion())) {
-            throwInvalidSamplingRuleManifestException("Manifest version: " + manifest.getVersion() + " is not supported.");
+    private static List<SamplingRule> processRuleManifest(SamplingRuleManifest manifest) {
+        SamplingRule defaultRule = manifest.getDefaultRule();
+        if (!SUPPORTED_VERSIONS.contains(manifest.getVersion())) {
+            throw newInvalidSamplingRuleManifestException("Manifest version: " + manifest.getVersion() + " is not supported.");
         }
-        if (null != defaultRule) {
-            if (null != defaultRule.getUrlPath() || null != defaultRule.getHost() || null != defaultRule.getHttpMethod()) {
-                throwInvalidSamplingRuleManifestException(
+        if (defaultRule != null) {
+            if (defaultRule.getUrlPath() != null || defaultRule.getHost() != null || defaultRule.getHttpMethod() != null) {
+                throw newInvalidSamplingRuleManifestException(
```

Comment:
I'm ok with using a factory for exceptions instead of custom to save a bit of code though both are good in different cases I guess. What isn't good is defining methods that only throw runtime exceptions, Java can't tell that it terminates code flow.

What do you find misleading here? The method creates a new exception without throwing it :)

Reply:
That's fair, definitely not defending how it was being done before I guess the factory way just looked weird to me. But you're right, a custom exception wouldn't really provide any value to the customer so this way is good.